node-backend/inProgressRepositoryService.ts

```diff
@@ -702,6 +691,36 @@ export function buildInProgressCopy(workspace: IOldWorkspace): IWorkspace {
   };
 }
 
+export function buildIdMap(
+  criteria: Record<string, IProblemCriterion>,
+  alternatives: Record<string, {title: string}>
+): Record<string, string> {
+  const criteriaIdMap = buildGenericIdMap(criteria);
+  const dataSourcesIdMap = buildDataSourcesIdMap(criteria);
```

Comment:
Very nice. Now you can do 
`buildGenericIdMap((criteria).flatMap('datasources').keyBy('id'))` ;)